pkg/store/mockstore/unistore/tikv/mock_region.go

```diff
@@ -513,6 +513,7 @@ func (rm *MockRegionManager) splitKeys(keys [][]byte) ([]*regionCtx, error) {
 
 		var i int
 		// Cannot be: i = range keys, probably the scope of i is different then
+		// nolint:intrange
```

Comment:
why not fix it in this PR 😂

Reply:
The [issue](https://go.dev/play/p/dDQr2Dm04vM) is that `for i = range N`, `i` will end up as N-1, vs `for i = 0; i < N; i++`, where i will end up as N. I will see how to fix it...

Reply:
So "fixing" it would make the code less readable :(
like it would be needed to add something like `fullRange` variable:
```
fullRange := true
for i = range keys {
  if ... {
    fullRange = false
    break
  }
}
if fullRange {
  i = len(keys)
}
splits := keys[:i]
keys = keys[i:]
```

Reply:
Hmm, I just moved the splits and keys into the loop instead, should work :)

Reply:
Oh I just though `for i = range N` is OK without seeing the end value problem 😂 However your really find an elegant way to fix it.